### ISSUE TITLE
Matplotlib 1.5.0 fails to build on Mac OS X 10.9 due to invalid setuptools version check...

### ISSUE THREAD

**Issue by bblacey** [Homebrew/homebrew-python#274]
The full failure log is available at https://travis-ci.org/bblacey/FreeCAD-MacOS-CI/builds/101292643

`==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/matplotlib/1.5.0/libexec --single-version-externally-managed --record=installed.txt
mock requires setuptools>=17.1. Aborting installation
/usr/local/Library/Homebrew/debrew.rb:11:in 'raise'
BuildError: Failed executing: python -c import\ setuptools,\ tokenize
__file__\ =\ 'setup.py'
exec(compile(getattr(tokenize,\ 'open',\ open)(__file__).read()
\ \ .replace('\r\n',\ '\n'),\ __file__,\ 'exec'))
 --no-user-cfg install --prefix=/usr/local/Cellar/matplotlib/1.5.0/libexec --single-version-externally-managed --record=installed.txt`

The changes introduced in moving to version 1.5.0 cause it to fail.  The previous 1.4.3 formula installed without errors.

**Comment by bblacey** [comment]
I just regressed this issue and it still fails.  Travis-ci log available at https://travis-ci.org/bblacey/FreeCAD-MacOS-CI/builds/103525127

**Comment by tdsmith** [comment]
Hmmm. I can leave this open but I can't reproduce the issue or explain why it's happening. I don't see this on a clean Mavericks VM.

**Comment by bblacey** [comment]
Thanks.  Happy to help troubleshoot...  The Travis-CI logs show the issue and if I simply install the 1.4.3 formula using the following, it works perfectly:

```
- brew tap --full homebrew/python
- cd /usr/local/Library/Taps/homebrew/homebrew-python/ && git checkout -q -f 1c4325b matplotlib.rb && cd ${TRAVIS_BUILD_DIR}
- brew install homebrew/python/matplotlib --with-pyside
```

FYI, I also noticed that matplotlib 1.5.1 is now available, superseding 1.5.0.

**Comment by tdsmith** [comment]
The issue is related to the newer version of mock that we're shipping and nothing to do with matplotlib specifically. The question is why system Python is seeing an older version of setuptools when it tries to install mock even though ours should take priority because it's in PYTHONPATH.

If there's an old setuptools egg somewhere, that might be the problem; can you try inserting `python -c 'import sys; print(sys.path)'` somewhere in the Travis configuration and seeing what the output looks like?

**Comment by bblacey** [comment]
Sure, will do.

**Comment by tdsmith** [comment]
Ugh, hilarious. The problem is that `/Library/Python/2.7/site-packages/setuptools-4.0.1-py2.7.egg` exists.

You can work around this by running `rm -rf /Library/Python/2.7/site-packages` as root before installing matplotlib (for clarity, it would make sense to run it early, since nothing you do depends on those files existing).

**Comment by dstufft** [comment]
https://github.com/pypa/pip/pull/3394

**Comment by tdsmith** [comment]
This should be fixed by Homebrew/homebrew@a239fb2c52ac5f87b14b85898503ba8baaeb3a1c, I think.
